applications/client/src/main/java/com/paritytrading/philadelphia/client/Messages.java

```diff
@@ -53,25 +53,38 @@ public void tooLowMsgSeqNum(FIXConnection connection, long receivedMsgSeqNum, lo
 
     @Override
     public void reject(FIXConnection connection, FIXMessage message) {
-        add(message);
+        messageCollection.add(message);
     }
 
     @Override
     public void logon(FIXConnection connection, FIXMessage message) {
-        add(message);
+        messageCollection.add(message);
     }
 
     @Override
     public void logout(FIXConnection connection, FIXMessage message) {
-        add(message);
+        messageCollection.add(message);
+    }
+}
+
+class MessageCollection {
```

Comment:
Adding an additional collection type is unwarranted and there is no need for synchronization.